### PR TITLE
Component unique fields

### PR DESCRIPTION
**Pull request by jhoward1994**
### What does it do?

Re implements component unique fields in V5

According to the logic:

- Component unique fields are validated against values in other entries of their parent content type in the same dimensions (locale and publication state)
- All unique values in a repeatable must be unique in the current entity
- The order of components in a repeatable does not matter. Validation will fail if any other entry (of content type + dimensions) holds the same value in the same component field anywhere in the repeatable

### Why is it needed?

Fix existing V4 bugs and implement core v5 functionality

### How to test it?

API tests
E2E tests

### Related issue(s)/PR(s)

DX-1348

WIP 

- [x] E2E tests
- [x] Update types for new CTs in API tests
- [x] Add repeatable cases to E2E

### CODE REVIEW

**Comment by jhoward1994** [comment]
Hey @innerdvations & @Bassel17 the WIPs are complete now so this is ready for review 🙏🏻

File: tests/api/core/strapi/document-service/resources/types/contentTypes.d.ts [hunk 1]
```diff
@@ -672,6 +672,8 @@ export interface ApiArticleArticle extends Struct.CollectionTypeSchema {
       'api::article.article'
     >;
     locale: Schema.Attribute.String;
+    idenitifers: Schema.Attribute.Component;
```
Comment (innerdvations) [inline tests/api/core/strapi/document-service/resources/types/contentTypes.d.ts:675]:
```suggestion
    identifiers: Schema.Attribute.Component;
```

File: tests/api/core/strapi/document-service/resources/types/contentTypes.d.ts [hunk 2]
```diff
@@ -672,6 +672,8 @@ export interface ApiArticleArticle extends Struct.CollectionTypeSchema {
...
+    repeatableIdenitifers: Schema.Attribute.Component;
```
Comment (innerdvations) [inline tests/api/core/strapi/document-service/resources/types/contentTypes.d.ts:676]:
```suggestion
    repeatableIdentifiers: Schema.Attribute.Component;
```

File: packages/core/types/src/modules/entity-validator.ts [hunk 3]
```diff
@@ -4,12 +4,10 @@ import type * as UID from '../uid';
 import type * as EntityService from './entity-service';
 
 type Entity = {
-  id: ID;
+  id: number;
```
Comment (innerdvations) [inline packages/core/types/src/modules/entity-validator.ts:7]: Are we sure this can't ever be string?

**Review by innerdvations — COMMENTED** [review]
Overall looks great! Just a few small comments

**Comment by innerdvations** [comment]
And API tests are failing, but that may be the issue on v5/main. Once we know they're fixed we can run them again.

File: packages/core/types/src/modules/entity-validator.ts (hunk 3 above)
Comment (jhoward1994) [reply packages/core/types/src/modules/entity-validator.ts:7]: Yes I think so. The entity validator works with entities not documents everywhere

**Review by innerdvations — APPROVED** [review]
LGTM!

**Review by innerdvations — APPROVED** [review]
I think we're basically there!

**Comment by jhoward1994** [comment]
Known issue with dynamic zones tracked here https://github.com/strapi/strapi/issues/20546

**Comment by jhoward1994** [comment]
This PR fixes the following issues in Strapi v5

closes https://github.com/strapi/strapi/issues/12541 
closes https://github.com/strapi/strapi/issues/12248
closes https://github.com/strapi/strapi/issues/10107
closes https://github.com/strapi/strapi/issues/17604
closes https://github.com/strapi/strapi/issues/16918

**Pull request merged**